backend/FwLite/MiniLcm.Tests/Validators/EntryValidatorTests.cs

```diff
@@ -11,24 +11,141 @@ public class EntryValidatorTests
+        var now = DateTime.UtcNow;
+        var entry = new Entry() { Id = entryId, DeletedAt = now, LexemeForm = new MultiString(){{"en", "lexeme"}}, Senses = [new Sense() { EntryId = Guid.Empty, }] };
+        _validator.TestValidate(entry).ShouldHaveValidationErrorFor("DeletedAt");
+    }
+
+    [Fact]
+    public void Succeeds_WhenLexemeFormIsPresent()
+    {
+        var entry = new Entry() { Id = Guid.NewGuid(), LexemeForm = new MultiString(){{"en", "lexeme"}} };
+        _validator.TestValidate(entry).ShouldNotHaveAnyValidationErrors();
+    }
+
+    [Fact]
+    public void Fails_WhenLexemeFormIsMissing()
+    {
+        var entry = new Entry() { Id = Guid.NewGuid() };
+        _validator.TestValidate(entry).ShouldHaveValidationErrorFor("LexemeForm");
+    }
+
+    [Fact]
+    public void Fails_WhenLexemeFormHasWsWithEmptyContent()
+    {
+        var entry = new Entry() { Id = Guid.NewGuid(), LexemeForm = new MultiString(){{"en", ""}} };
+        _validator.TestValidate(entry).ShouldHaveValidationErrorFor("LexemeForm");
+    }
+
+    [Theory]
+    [InlineData(nameof(Entry.CitationForm))]
+    [InlineData(nameof(Entry.LiteralMeaning))]
+    [InlineData(nameof(Entry.Note))]
+    public void Succeeds_WhenNonEmptyFieldIsPresent(string fieldName)
+    {
+        var entry = new Entry() { Id = Guid.NewGuid(), LexemeForm = new MultiString(){{"en", "lexeme"}} };
+        SetProperty(entry, fieldName, "content");
+        _validator.TestValidate(entry).ShouldNotHaveAnyValidationErrors();
+    }
+
+    [Theory]
+    [InlineData(nameof(Entry.CitationForm))]
+    [InlineData(nameof(Entry.LiteralMeaning))]
+    [InlineData(nameof(Entry.Note))]
+    public void Succeeds_WhenNonEmptyFieldHasNoContent(string fieldName)
+    {
+        var entry = new Entry() { Id = Guid.NewGuid(), LexemeForm = new MultiString(){{"en", "lexeme"}} };
+        MakePropertyEmpty(entry, fieldName);
+        _validator.TestValidate(entry).ShouldNotHaveAnyValidationErrors();
+    }
+
+    [Theory]
+    [InlineData(nameof(Entry.CitationForm))]
+    [InlineData(nameof(Entry.LiteralMeaning))]
+    [InlineData(nameof(Entry.Note))]
+    public void Fails_WhenNonEmptyFieldHasWsWithEmptyContent(string fieldName)
+    {
+        var entry = new Entry() { Id = Guid.NewGuid(), LexemeForm = new MultiString(){{"en", "lexeme"}} };
+        SetProperty(entry, fieldName, "");
+        _validator.TestValidate(entry).ShouldHaveValidationErrorFor(fieldName);
+    }
+
+    [Fact]
+    public void Fails_WhenComplexFormsContainCircularReference()
+    {
+        var entryId = Guid.NewGuid();
+        var entry = new Entry() { Id = entryId, LexemeForm = new MultiString(){{"en", "lexeme"}}, ComplexForms = [new ComplexFormComponent(){ ComplexFormEntryId = entryId, ComponentEntryId = Guid.Empty }] };
+        _validator.TestValidate(entry).ShouldHaveValidationErrorFor("ComplexForms[0]");
+        // _validator.TestValidate(entry).ShouldNotHaveAnyValidationErrors();
```

Comment:
you can probably remove this commented code

Reply:
Done.